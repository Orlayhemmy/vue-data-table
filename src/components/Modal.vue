<template>
  <div id='modal-component'>
    <button @click='toggleModal()'>
      {{title}}
    </button>
    <div v-if='isVisible'  class='modal-content'>
      <span v-show='isVisible' @click='toggleModal()' class='close-modal'>&#x2715;</span>
      <slot></slot>
      <div>
      <button v-show='isVisible' @click='handleSubmit()'> {{actionButtonText}} </button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'modal-component',
  props: ['actionButtonText', 'title', 'handleSubmit'],
  data() {
    return {
      isVisible: false,
    }
  },
  methods: {
    toggleModal() {
      return this.isVisible = !this.isVisible
    },
  },
}
</script>
<style lang='scss' scoped>
@import '../assets/styles/modal.scss';
</style>
