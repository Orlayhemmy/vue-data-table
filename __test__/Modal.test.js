import Vue from 'vue';
import Modal from '../src/components/Modal';
import { mount } from '@vue/test-utils';

describe('Modal Component', () => {
  const wrapper = mount(Modal, {
    propsData: {
      title: "Add Transaction",
      actionButtonText: 'Submit',
    },
    slots: {
      default: `<input type="text" />`
    }
  });

  it('renders correctly without errors', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should display the appropriate title for the modal button', () => {
    const title = wrapper.find('button');
    expect(title.text()).toContain('Add Transaction')
  });

  it('should change isVisible to true and open modal when button is clicked', () => {
    expect(wrapper.vm.isVisible).toBe(false);
    expect(wrapper.find('.modal-content').exists()).toBe(false);


    const button = wrapper.find('button');
    button.trigger('click');

    expect(wrapper.vm.isVisible).toBe(true);
    expect(wrapper.find('.modal-content').exists()).toBe(true);
  });

  it('should display the content in the slot when modal opens', () => {
    expect(wrapper.find('.modal-content').findAll('input').length).toBe(1);
  });

  it('should change isVisible to false and close the modal', () => {
    expect(wrapper.vm.isVisible).toBe(true);

    const closeButton = wrapper.find('.close-modal');
    closeButton.trigger('click');

    expect(wrapper.vm.isVisible).toBe(false);
  });
})