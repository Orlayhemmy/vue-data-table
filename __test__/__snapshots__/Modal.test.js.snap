// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correctly without errors 1`] = `
<div id="modal-component"><button>
    Add Transaction
  </button>
  <!---->
</div>
`;
